<?xml version="1.0"?>

<launch>
    <!-- Launch the message collector -->
    <node pkg="ros_kafka_connector" type="msg_collector" name="msg_collector" output="screen">
    </node>

    <node pkg="ros_kafka_connector" type="object_publish.py" name="object_publish">
        <param name="~bootstrap_server" value="114.70.21.162:9092"/>
        <param name="~kafka_topic" value="object"/>
        <param name="~ros_topic" value="/grid/obj"/>
        <param name="~msg_type" value="ObjectInfo"/>
    </node>

    <node pkg="ros_kafka_connector" type="vehicle_publish.py" name="vehicle_publish">
        <param name="~bootstrap_server" value="114.70.21.162:9092"/>
        <param name="~kafka_topic" value="vehicle"/>
        <param name="~ros_topic" value="/vehicle"/>
        <param name="~msg_type" value="VehicleInfo"/>
    </node>
</launch>
